src/Http.ts

```diff
@@ -118,13 +127,17 @@ export function post<A>(url: string, body: unknown, decoder: Decoder<A>): Reques
 }
 
 // --- Helpers
-function xhr<A>(req: Request<A>): Observable<Either<HttpError, A>> {
+function xhrWithResp<A>(req: Request<A>): Observable<Either<HttpError, Response<A>>> {
```

Comment:
@bmazzarol I think that we can keep only the `xhr` function with a different signature:
```ts
function xhr<A>(req: Request<A>): Observable<Either<HttpError, Response<A>>>
```

and then handle the right side differently in `send()` and `sendWithResp()`:
```ts
type XHR<A> = Observable<Either<HttpError, Response<A>>>

export function toTask<A>(req: Request<A>): TaskEither<HttpError, A> {
  return () => toResult(xhr(req)).toPromise()
}

export function send<A, Msg>(f: (e: Either<HttpError, A>) => Msg): (req: Request<A>) => Cmd<Msg> {
  return req => toResult(xhr(req)).pipe(map(result => T.of(O.some(f(result)))))
}

export function sendWithResp<A, Msg>(f: (e: Either<HttpError, Response<A>>) => Msg): (req: Request<A>) => Cmd<Msg> {
  return req => xhr(req).pipe(map(result => T.of(O.some(f(result)))))
}

// --- Helpers
function toResult<A>(x: XHR<A>): Observable<Either<HttpError, A>> {
  return x.pipe(map(E.map(resp => resp.body)))
}

function xhr<A>(req: Request<A>): XHR<A> {
  return ajax(toXHRRequest(req)).pipe(
    map(flow(toResponse(req), decodeWith(req.expect))),
    catchError((e: unknown): XHR<A> => of(E.left(toHttpError(req, e))))
  )
}

function decodeWith<A>(decoder: Decoder<A>): (resp: Response<string>) => Either<HttpError, Response<A>> {
  return resp =>
    pipe(
      // By spec parsing json can only throw `SyntaxError`
      E.parseJSON(resp.body, e => (e as SyntaxError).message),
      E.chain(decoder),
      E.bimap(
        e => ({
          _tag: 'BadPayload',
          value: e,
          response: resp
        }),
        body => ({ ...resp, body })
      )
    )
}
```

Reply:
Sure will do.